hw-4/controllers/groups.controller.ts

```diff
@@ -0,0 +1,51 @@
+import { Request, Response } from 'express';
+
+import Group from '@app-models/group';
+import GroupRepository from '../data-access/group.repository';
+import GroupService from '../services/groups.service';
+
+const groupService = new GroupService(new GroupRepository());
+
+export const getGroups = (req: Request, res: Response): void => {
+  groupService.getGroups().then((groups) => res.json(groups));
+};
+
+export const getGroup = (req: Request, res: Response): void => {
+  groupService.getGroupById(req.params.id).then((group) => res.json(group));
```

Comment:
We should follow a pattern on how we get the values from the `req` object.

So, let's review the usage of `body` and `params` here in this class to always destructure it directly from the argument list:

```typescript
export const getGroup = ({ body, params }: Request, res: Response): void => {
   const { id } = params;
```

Reply:
Done here and in users.controller as well.